pyproject.toml

```diff
@@ -1,2 +1,58 @@
 [tool.black]
 target-version = ['py310']
+
+[tool.poetry]
+name = "screenpy"
+version = "4.0.2"
+description = "Screenplay pattern base for Python automated UI test suites."
+authors = ["Perry Goy <perry.goy@gmail.com>"]
+license = "MIT"
+#packages = []
+repository = "https://github.com/ScreenPyHQ/screenpy"
+documentation = "https://screenpy-docs.readthedocs.io"
+readme = "README.md"
+classifiers = [
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3 :: Only",
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Quality Assurance",
+    "Topic :: Software Development :: Testing",
+    "Topic :: Software Development :: Testing :: BDD",
+    "Natural Language :: English",
+    "License :: OSI Approved :: MIT License",
+]
+
+[tool.poetry.dependencies]
+python = "^3.10"
+PyHamcrest = "~2.0"  # ">=2.0.0,<2.1"
+typing-extensions = ">=4.1.1,<4.4"
+
+[tool.poetry.dev-dependencies]
+pre-commit = "^2.17.0"
+pytest = "^7.1.1"
+pylint = "^2.12.2"
+mypy = "^0.941"
+black = "^22.1.0"
+isort = "^5.10.1"
+flake8 = "^4.0.1"
+coverage = "*"
+
+[tool.poetry.extras]
+allure = ["screenpy_adapter_allure"]
+appium = ["screenpy_appium"]
+playwright = ["screenpy_playwright"]
+pyotp = ["screenpy_pyotp"]
+requests = ["screenpy_requests"]
+selenium = ["screenpy_selenium"]
+
+[build-system]
+requires = ["poetry-core>=1.2.0"]
+build-backend = "poetry.core.masonry.api"
+
+#[build-system]
+#requires = [
+#    "setuptools==63.4.3",
+#    "wheel",
+#]
+#build-backend = "setuptools.build_meta"
```

Comment:
Temporarily leaving this comment in for discussion purposes.

Reply:
Let's discuss!

What does this do?